### PR TITLE
Fix error with "None" key press

### PR DESCRIPTION
**Pull request by brisvag**
# Description
Unfortunately I struggle to reproduce the error which I got during normal use, but one way to trigger it is to quickly close napari as soon as it opens:

```py
File ~/venv/test-napari/lib/python3.10/site-packages/napari/utils/key_bindings.py:520, in KeymapHandler.on_key_press(self=<napari.utils.key_bindings.KeymapHandler object>, event=<KeyEvent blocked=False handled=False key=None m...d80> source=None sources=[] text= type=key_press>)
    511 def on_key_press(self, event):
    512     """Called whenever key pressed in canvas.
    513
    514     Parameters
   (...)
    517         The vispy key press event that triggered this method.
    518     """
    519     combo = normalize_key_combo(
--> 520         components_to_key_combo(event.key.name, event.modifiers)
        event = <KeyEvent blocked=False handled=False key=None modifiers=() native=<PyQt5.QtGui.QKeyEvent object at 0x7f64380e1d80> source=None sources=[] text= type=key_press>
    521     )
    523     repeatables = {
    524         *action_manager._get_repeatable_shortcuts(self.keymap_chain),
    525         "Up",
   (...)
    528         "Right",
    529     }
    531     if (
    532         event.native is not None
    533         and event.native.isAutoRepeat()
   (...)
    537         # unless the combo being held down is one of the autorepeatables or
    538         # one of the navigation keys (helps with scrolling).

AttributeError: 'NoneType' object has no attribute 'name'
```

I'm not entirely sure this is the right fix, but it seems like a reasonable thing to do, and shouldn't be that expensive.

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/developers/translations.html).

### CODE REVIEW

File: napari/utils/key_bindings.py [hunk 1]
```diff
@@ -516,6 +516,9 @@ def on_key_press(self, event):
         event : vispy.util.event.Event
             The vispy key press event that triggered this method.
         """
+        if event.key is None:
+            return
```
Comment (Czaki) [inline napari/utils/key_bindings.py]:
```suggestion
        if event.key is None:
            # TODO determine when None key could be sent. 
            return
```

**Review by Czaki — APPROVED** [review]
As You mention it is not clear if it is a proper place to solve this problem then it should be reflected not only in the PR description but also in comments in the code. I add some simple propositions but maybe you will have a better idea.

**Review by goanpeca — APPROVED** [review]
Looks good!

**Review by alisterburt — APPROVED** [review]
Looks good! merging

**Comment by Czaki** [comment]
@alisterburt hm. @andy-sweet is the release manager, and he should do this to peace updates of the changelog.

**Comment by andy-sweet** [comment]
> @alisterburt hm. @andy-sweet is the release manager, and he should do this to peace updates of the changelog.

Well technically, we're [joint release managers](https://napari.zulipchat.com/#narrow/stream/215289-release/topic/0.2E4.2E17/near/297079669).

Not sure if that's a good idea for future releases, but @alisterburt you should feel free to merge things as you see fit.

**Pull request merged**